### PR TITLE
image generation with ball smoothing rather than exponential propagation

### PR DESCRIPTION
**Pull request by charlio23**
Balls are now smoothed circles rather than a propagated exponential.

### CODE REVIEW

File: environments/chaotic_pendulum.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import cv2
```
Comment (OleguerCanal) [inline environments/chaotic_pendulum.py:1]: Add cv2 to requirements.txt

File: environments/environment.py [hunk 2]
```diff
@@ -76,6 +76,13 @@ def get_default_radius_bounds(self):
         """
         raise NotImplementedError
 
+    def _world_to_pixels(self, x, y, res):
```
Comment (OleguerCanal) [inline environments/environment.py:79]: Nice function! Add docstring.

**Review by OleguerCanal — CHANGES_REQUESTED** [review]
Very nice improvement!

File: environments/chaotic_pendulum.py (hunk 1 above)
Comment (charlio23) [reply environments/chaotic_pendulum.py:1]: sorryy

File: environments/environment.py (hunk 2 above)
Comment (charlio23) [reply environments/environment.py:79]: :)

File: environments/environment.py [hunk 3]
```diff
@@ -77,7 +77,16 @@ def get_default_radius_bounds(self):
...
     def _world_to_pixels(self, x, y, res):
+        """Maps coordinates from world space to pixel space
 
+        Args:
+            x (float): x coordinate of the world space.
+            y (float): y coordinate of the world space.
+            res (int): Image resolution in pixels (images are square).
+
+        Returns:
+            (float, float): Tuple of coordinates in pixel space.
```
Comment (OleguerCanal) [inline environments/environment.py]:
```suggestion
            (int, int): Tuple of coordinates in pixel space.
```

**Pull request merged**